### ISSUE TITLE
Fancylists to package MarkBind's current and future list features

### ISSUE THREAD

**Issue by KevinEyo1** [MarkBind/markbind#2458]
### Please confirm that you have searched existing issues in the repo

Yes, I have searched the existing issues

### Any related issues?

#2351, #2405

### What is the area that this feature belongs to?

Plugins

### Is your feature request related to a problem? Please describe.

With the amount of new list features that can bring about greater flexibility for users, it might be better to package it all together as a MarkBind component.

### Describe the solution you'd like

A new MarkBind component that will house all MarkBind exclusive list features

**Comment by yucheng11122017** [comment]
Hi @KevinEyo1, I'm not sure what is the difference between what we currently have and what you are proposing. Since the list features are all kind of together already. 
Could you clarify on what you hope to achieve with this issue?

**Comment by KevinEyo1** [comment]
As per discussion with @damithc, I think it is to abstract out the extra functionalities under our own special component for those looking for greater customization and flexibility compared to a normal list, especially for the more niche features.

**Comment by kaixin-hc** [comment]
I still don't understand, can you explain how this would change functionality for users and development for developers? Thanks!

**Comment by damithc** [comment]
@kaixin-hc The list customization is both a significant and a unique feature we have but it is currently buried under list formatting which is a commonplace feature that every static site generator has. People wouldn't even know about it unless they specifically started looking for such a feature. I think we can sell it as a separate feature with its own name.

**Comment by yucheng11122017** [comment]
> @kaixin-hc The list customization is both a significant and a unique feature we have but it is currently buried under list formatting which is a commonplace feature that every static site generator has. People wouldn't even know about it unless they specifically started looking for such a feature. I think we can sell it as a separate feature with its own name.

Ah ok so its just a repackaging of our existing functionality?

**Comment by damithc** [comment]
> Ah ok so its just a repackaging of our existing functionality?

Yup, just for appearances. No change to the implementation at all.

**Comment by yucheng11122017** [comment]
> > Ah ok so its just a repackaging of our existing functionality?
> 
> Yup, just for appearances. No change to the implementation at all.

HI @damithc, does it mean this is just a documentation change? Or do you want to add syntactic sugar?

**Comment by damithc** [comment]
> HI @damithc, does it mean this is just a documentation change? Or do you want to add syntactic sugar?

@yucheng11122017 Just a documentation change. 

The name is up for debate as well. 'Fancy lists' is just the first name that came to my mind. Just did a search for 'Markdown fancy lists' and this came up https://github.com/Moxio/markdown-it-fancy-lists
